source/val/validate_memory.cpp

```diff
@@ -427,6 +427,27 @@ spv_result_t ValidateVariable(ValidationState_t& _, const Instruction* inst) {
       }
     }
   }
+  // Vulkan Appendix A: Validation Rules within a Module
+  // Check when VK_KHR_shader_float16_int8 is enabled but
+  // VK_KHR_8bit_storage or VK_KHR_16bit_storage are not.
+  if (_.IncludesIntOrFloatOfWidth(result_type->word(3), SpvOpTypeInt, 8) &&
```

Comment:
SPIR-V capabilities are not a direct mapping for the Vulkan functionality. In SPIR-V, 8-bit storage is accessible from the 8-bit storage capabilities *or* the Int8 capability. Similar for half floats. Some work likely belongs in the validation layers to check that the right Vulkan feature bits are enabled, but in SPIR-V if the Int8 capability is present, that is sufficient to be able to store and use 8-bit integers.

Reply:
Hi @alan-baker , we merged https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/514/commits/c6c1561a632eda94c86fd5cb7d8b3aeea1ffca46 , would that be enough coverage for this extension?

Reply:
Probably if only because adding more stringent checks will be quite difficult. The validation layers would have to inspect the SPIR-V with respect to the Vulkan feature bits. If there were any load/store containing an 8 or 16 bit type then the associated storage feature bit in Vulkan would have to be set. This would be fairly awkward to put into the validation layers is my understanding.

Reply:
Agreed. I will close this PR then, thank you very much for your help!